src/status_im/contexts/settings/wallet/effects.cljs

```diff
@@ -0,0 +1,45 @@
+(ns status-im.contexts.settings.wallet.effects
+  (:require [native-module.core :as native-module]
+            [promesa.core :as promesa]
+            [status-im.contexts.syncing.events :as syncing-events]
+            [status-im.contexts.syncing.utils :as sync-utils]
+            [utils.re-frame :as rf]
+            [utils.security.core :as security]
+            [utils.transforms :as transforms]))
+
+(rf/reg-fx :effects.connection-string/export-keypair
+ (fn [{:keys [key-uid sha3-pwd keypair-key-uid on-success on-fail]}]
+   (let [config-map (transforms/clj->json {:senderConfig {:loggedInKeyUid   key-uid
+                                                          :keystorePath     ""
+                                                          :keypairsToExport [keypair-key-uid]
+                                                          :password         (security/safe-unmask-data
+                                                                             sha3-pwd)}
+                                           :serverConfig {:timeout 0}})]
+     (-> (native-module/get-connection-string-for-exporting-keypairs-keystores
+          config-map)
+         (promesa/then (fn [response]
+                         (if (sync-utils/valid-connection-string? response)
+                           (on-success response)
+                           (on-fail (throw (js/Error.
+                                            "generic-error: failed to get connection string"))))))
```

Comment:
I wouldn't recommend throwing a JS error as this might crash the app. Instead, we handle gracefully with a toast or logs.

Reply:
I think here it's also incorrect. `on-fail` should receive an error of some sort (string, even an instance of js/Error) but we shouldn't throw.

Reply:
can we say that `native-module/get-connection-string-for-exporting-keypairs-keystores` always return valid connection string on `promesa/then` ?

Reply:
Yes :) If we receive any error from status-go, the `promisify-callback` gracefully rejects the promise for us. Sorry for the vague comment earlier.